nixos/tests/clatd.nix

```diff
@@ -26,229 +26,228 @@
+    # server, the client starts the clat daemon which starts and configures the
+    # local IPv4 -> IPv6 translation via Tayga after discovering the PLAT
+    # prefix via DNS64.
+    client = {
+      virtualisation.vlans = [
+        3 # towards router
+      ];
 
-        systemd.network = {
-          enable = true;
-          networks."vlan1" = {
-            matchConfig.Name = "eth1";
+      networking = {
+        useDHCP = false;
+        interfaces.eth1 = lib.mkForce { };
+      };
 
-            # NOTE: clatd does not actually use the PREF64 prefix discovered by
-            # systemd-networkd (nor does systemd-networkd do anything with it,
-            # yet), but we set this to confirm it works. See the test script
-            # below.
-            ipv6AcceptRAConfig.UsePREF64 = true;
-          };
-        };
+      systemd.network = {
+        enable = true;
+        networks."vlan1" = {
+          matchConfig.Name = "eth1";
 
-        services.clatd = {
-          enable = true;
-          # NOTE: Perl's Net::DNS resolver does not seem to work well querying
-          # for AAAA records to systemd-resolved's default IPv4 bind address
-          # (127.0.0.53), so we add an IPv6 listener address to systemd-resolved
-          # and tell clatd to use that instead.
-          settings.dns64-servers = "::1";
+          # NOTE: clatd does not actually use the PREF64 prefix discovered by
+          # systemd-networkd (nor does systemd-networkd do anything with it,
+          # yet), but we set this to confirm it works. See the test script
+          # below.
+          ipv6AcceptRAConfig.UsePREF64 = true;
         };
+      };
 
-        # Allow clatd to find dns server. See comment above.
-        services.resolved.extraConfig = ''
-          DNSStubListenerExtra=::1
-        '';
-
-        environment.systemPackages = [ pkgs.mtr ];
+      services.clatd = {
+        enable = true;
+        # NOTE: Perl's Net::DNS resolver does not seem to work well querying
+        # for AAAA records to systemd-resolved's default IPv4 bind address
+        # (127.0.0.53), so we add an IPv6 listener address to systemd-resolved
+        # and tell clatd to use that instead.
+        settings.dns64-servers = "::1";
       };
+
+      # Allow clatd to find dns server. See comment above.
+      services.resolved.extraConfig = ''
+        DNSStubListenerExtra=::1
+      '';
+
+      environment.systemPackages = [ pkgs.mtr ];
     };
+  };
 
-    testScript = ''
-      import json
+  testScript = ''
+    import json
 
-      start_all()
+    start_all()
 
-      # wait for all machines to start up
-      for machine in client, router, server:
-        machine.wait_for_unit("network-online.target")
+    # wait for all machines to start up
+    for machine in client, router, server:
+      machine.start_job("network-online.target")
+      machine.wait_for_unit("network-online.target")
```

Comment:
Can we wait for network.target instead?

Reply:
Yes, that works. Should I still start `network-online.target`?